File: core/src/commonMain/kotlin/com/xhsdn/core/naming/NamingFormat.kt
package com.xhsdn.core.naming

/**
 * 命名模板占位符常量与默认模板。
 * 与 [app/.../NamingFormat.java] 保持一致。
 */
object NamingFormat {
    const val TOKEN_USERNAME = "username"
    const val TOKEN_USER_ID = "user_id"
    const val TOKEN_TITLE = "title"
    const val TOKEN_POST_ID = "post_id"
    const val TOKEN_PUBLISH_TIME = "publish_time"
    const val TOKEN_INDEX = "index"
    const val TOKEN_INDEX_PADDED = "index_padded"
    const val TOKEN_DOWNLOAD_TIMESTAMP = "download_timestamp"

    const val DEFAULT_TEMPLATE = "{post_id}_{index_padded}"

    fun buildPlaceholder(token: String): String = "{$token}"
}

File: web/src/core/naming/NamingFormat.ts
/**
 * 命名模板占位符常量与默认模板。
 * 1:1 翻译自 `core/src/commonMain/kotlin/com/xhsdn/core/naming/NamingFormat.kt`。
 */

export const TOKEN_USERNAME = 'username';
export const TOKEN_USER_ID = 'user_id';
export const TOKEN_TITLE = 'title';
export const TOKEN_POST_ID = 'post_id';
export const TOKEN_PUBLISH_TIME = 'publish_time';
export const TOKEN_INDEX = 'index';
export const TOKEN_INDEX_PADDED = 'index_padded';
export const TOKEN_DOWNLOAD_TIMESTAMP = 'download_timestamp';

export const DEFAULT_TEMPLATE = `{${TOKEN_POST_ID}}_{${TOKEN_INDEX_PADDED}}`;

export function buildPlaceholder(token: string): string {
  return `{${token}}`;
}
